neurons/validator.py

```diff
@@ -571,7 +584,8 @@ async def run_step(self):
                 hotkey
             )
 
-            losses = [math.inf for _ in range(len(batches))]
+            old_losses = [math.inf for _ in range(len(old_batches))]
+            new_losses = [math.inf for _ in range(len(new_batches))]
```

Comment:
I do not think there is a new to defined two loss variables. they both have the same length with `math.inf` values

Reply:
I wrote it like this because old_batches and new_batches will probably have different lengths, since we are using different documents for each batch.

For example:
indices_of_pages_we_use_from_falcon = [0, 5, 10]
indices_of_pages_we_use_from_fineweb = [2, 4, 8]

With the current SubsetLoader (buffer style) we will have buffers of different sizes, and therefore the one with the biggest buffer will have more batches.

Whenever we start using the new logic (one page --> one batch), then yes, both old and new batches should have the same length.

Reply:
You are right!